### DISCUSSION TITLE
Ubuntu 18.04 Support

### DISCUSSION THREAD

**Discussion by ryanwwest** [srsran/srsRAN_Project#59] [General Help] [answered]
I was wondering if srsRAN can  be built and installed on Ubuntu 18.04? I already have UHD installed and followed the dependency installation steps (including getting the 9.4 version of GCC) but when I run `sudo make -j $(nproc)` it errors out as follows:

```
[ 17%] Building CXX object lib/srsvec/CMakeFiles/srsvec.dir/conversion.cpp.o
In file included from /opt/srsRAN_Project/lib/asn1/asn1_utils.cpp:23:0:
/opt/srsRAN_Project/include/srsran/asn1/asn1_utils.h:809:40: error: no matching function for call to ‘max_size(<brace-enclosed initializer list>)’
   ext_array<uint8_t, max_size({UB, 64U})> octets_;
... < many more errors>
```

I'd like to get this on 18.04 if possible, and I'm unsure if this error would be encountered on 22.04 as well. I was wondering if anyone can give any tips for getting past these errors (or alternatively, if there is a way to download a prebuilt version and just run it)?

**Comment by brendan-mcauliffe** [comment, marked as answer]
Hi Ryan, srsRAN Project requires at least Ubuntu 20.04 and gcc v9.4 (or clang v10). We recommend Ubuntu 22.04 and gcc v9.4/ clang v10 or later. You can see more on the requirements [here](https://docs.srsran.com/projects/project/en/latest/user_manuals/source/installation.html#manual-installation) in the docs

We currently do not plan to add support for older versions of Ubuntu and gcc/clang.
